ProcessLib/HydroMechanics/HydroMechanicsFEM.h

```diff
@@ -0,0 +1,425 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESS_LIB_HYDROMECHANICS_FEM_H_
+#define PROCESS_LIB_HYDROMECHANICS_FEM_H_
+
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include "MaterialLib/SolidModels/KelvinVector.h"
+#include "MaterialLib/SolidModels/LinearElasticIsotropic.h"
+#include "MathLib/LinAlg/Eigen/EigenMapTools.h"
+#include "NumLib/Fem/FiniteElement/TemplateIsoparametric.h"
+#include "NumLib/Fem/ShapeMatrixPolicy.h"
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Deformation/LinearBMatrix.h"
+#include "ProcessLib/LocalAssemblerInterface.h"
+#include "ProcessLib/LocalAssemblerTraits.h"
+#include "ProcessLib/Parameter/Parameter.h"
+#include "ProcessLib/Utils/InitShapeMatrices.h"
+
+#include "HydroMechanicsProcessData.h"
+#include "LowerDimShapeTable.h"
+
+namespace ProcessLib
+{
+namespace HydroMechanics
+{
+template <typename BMatricesType, typename ShapeMatrixTypeDisplacement,
+          typename ShapeMatricesTypePressure, int DisplacementDim>
+struct IntegrationPointData final
+{
+    explicit IntegrationPointData(
+        MaterialLib::Solids::MechanicsBase<DisplacementDim>& solid_material)
+        : _solid_material(solid_material),
+          _material_state_variables(
+              _solid_material.createMaterialStateVariables())
+    {
+    }
+
+#if defined(_MSC_VER) && _MSC_VER < 1900
+    // The default generated move-ctor is correctly generated for other
+    // compilers.
+    explicit IntegrationPointData(IntegrationPointData&& other)
+        : _b_matrices(std::move(other._b_matrices)),
+          _sigma(std::move(other._sigma)),
+          _sigma_prev(std::move(other._sigma_prev)),
+          _eps(std::move(other._eps)),
+          _eps_prev(std::move(other._eps_prev)),
+          _solid_material(other._solid_material),
+          _material_state_variables(std::move(other._material_state_variables)),
+          _C(std::move(other._C)),
+          integration_weight(std::move(other.integration_weight))
+    {
+    }
+#endif  // _MSC_VER
+
+    typename ShapeMatrixTypeDisplacement::NodalRowVectorType _N_u;
+    typename BMatricesType::BMatrixType _b_matrices;
+    typename BMatricesType::KelvinVectorType _sigma, _sigma_prev;
```

Comment:
the stresses here are actually effective stress. please rename the variables such that we can easily distinguish them from total stress.